### DISCUSSION TITLE
New pricing and breaking changes!

### DISCUSSION THREAD

**Discussion by mikhail-tokarev** [codemagic-ci-cd/codemagic-docs#933] [Announcements]
As of **September 12, 2022**, our default machine type will be **Mac mini M1** instead of the Intel-based Mac mini. This change comes together with the release of our new Pay as you go plan that lowers the price for M1 machines by half and introduces a $299/month cap for unlimited build minutes, read more about it [here](https://blog.codemagic.io/new-pricing/).

As a result, you should be aware of the following changes:

- If you haven’t specified an instance type in your codemagic.yaml, your builds will run on Mac mini M1 starting from September 12. Personal accounts will have 500 free M1 minutes every month.

- All Mac mini M1 have Xcode 13.3.1 and later available. To use an older version of Xcode, switch to an Intel-based macOS machine (`mac_mini` or `mac_pro` instance type).

- The current version of Apple M1 virtualization does not support Android emulator or Docker containers. If your build **requires Android emulator or Docker**, change your workflow to run on an Intel-based macOS, Linux or a Windows machine.

To keep using Intel-based Mac mini, set `instance_type` to `mac_mini`. As the old Mac mini machines are being discontinued, we no longer provide a free access to `mac_mini` instance type starting from September 12.

**Comment by soygcm** [comment]
So basically if I'm paying and I haven't specified `mac_mini` my pipeline just brake. Why you didn't keep the `mac_mini` by default for pay users?

**Reply by mikhail-tokarev** [reply]
Hi @soygcm 
not really. If you on pay-as-you-go plan and don't have instance type in `codemagic.yaml` file your build will be run using `mac_mini_m1` instance type. And you can switch back to any other instance type by explicitly adding it to the yaml file.

**Comment by brightmind-loredo** [comment]
I only use the build UI so I don't have a build yaml.  I changed to m1 in my build settings and built again.  It took the same time as before.  The build summary says "Machine: Mac Pro", build ID: 632cd35c3dad27b532d926fd ... Am I missing something?

**Reply by mikhail-tokarev** [reply]
Hi @brightmind-loredo 
thank you for providing build id, I can see that it indeed used Mac Pro machine. Is there any chance you changed M1 in wrong workflow?

**Comment by gslender** [comment]
I can only see macOS M1 Standard VM in the workflow, and it seems since adding a billing account details (ie upgrading to pay as I go), I don't get 500 free M1 mins, so maybe I should downgrade back again. Are you supposed to still get 500 mins free even if you're enabled for billing?

**Reply by mikhail-tokarev** [reply]
Hi @gslender 

thanks for your question. Yes, all personal accounts still receive 500 free minutes every month and can use pay-as-you-go model if you require more build minutes. And if you created a team account you need to pay for all consumed build minutes (but never more than $299/mo). Let me know if you have further questions

**Reply by gslender** [reply]
The billing page shows that the macOS M1 Standard VM minutes are increasing and charges are being applied and no free m1 minutes are being spent - I don't think it's working the way you have suggested 

So far, every time I execute a workflow, it charges my account

**Reply by gslender** [reply]
My account is personal too

**Reply by mikhail-tokarev** [reply]
can you share any build id? we will take a close look

**Reply by gslender** [reply]
6334fe236e6d538fde880540

**Reply by mikhail-tokarev** [reply]
we've checked your usage history and can see you used M1 minutes before September 12 when it was a paid feature. Right now you are using free minutes, so everything looks correct. Anyway feel free to contact us if you have any further questions!

**Comment by Alvaro948** [comment]
Hi @mikhail-tokarev, after setting instance type from `mac_mini` to `mac_mini_m1` the build is failing on ionic capacitor:

```yaml
      - name: Build ionic app
        script: |
          ionic cap build android --configuration production
```

![image](https://user-images.githubusercontent.com/97282850/199691385-69d9c5df-1eab-48fb-8e7e-60fd5e5b8603.png)

Failed build: 63638b8db83454e5368c79ca
Successful build: 63638c6a49418626798bb6de (after reverting the instance type back to `mac_mini`)

![image](https://user-images.githubusercontent.com/97282850/199691165-647fb4fe-5df1-4c42-a96f-2014cf2a63d0.png)

**Comment by HCDevSG** [comment]
![Screenshot 2023-05-24 at 10 34 19 AM](https://github.com/codemagic-ci-cd/codemagic-docs/assets/134362745/015b1d42-820f-457c-ab91-61655eee4de4)


I did not used the code magic free minutes more then 6 minutes but I'm getting this issue
